.github/workflows/invalidatePRs.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+# GitHub workflow handles invalidating Pull Requests after 3 hours.
+#
+# It will only invalidate open, non-draft pull requests and mark each status
+# as timed out.
+
+if [[ "" == $token ]]; then
+  echo Missing the environment variable "$token"
+  exit
+fi
+
+oauth=$token
+shouldDebug=false
+
+if [[ "" != $debug ]]; then
+  shouldDebug=true
+  echo Initializing debug log level
+fi
+
+# Get all active PRs
+# https://docs.github.com/en/rest/reference/pulls#list-pull-requests
+prs=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $oauth" \
+  https://api.github.com/repos/imodeljs/imodeljs/pulls)
+
+_jq() {
+  echo ${1} | base64 --decode | jq -r ${2}
+}
+
+function log() {
+  if [[ $shouldDebug == true  ]]; then
+    echo "${1}"
+  fi
+}
+
+declare -A listSha
+
+# Get head sha on master branch
+masterCommit=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $oauth" \
+  https://api.github.com/repos/imodeljs/imodeljs/commits/master)
+
+masterHeadCommit=$(echo ${masterCommit} | jq -r '. | @base64')
+masterSha=$(_jq  ${masterHeadCommit} '.sha')
+
+listSha["master"]=${masterSha}
+echo "Master SHA is ${listSha["master"]}"
+
+for pr in $(echo "${prs}" | jq -r '.[] | @base64'); do
+  log "$(_jq ${pr} '.title') (#$(_jq ${pr} '.number'))"
+
+  if [[ "open" != $(_jq ${pr} '.state') ]] || [[ "true" == $(_jq ${pr} '.draft') ]]; then
+    log "  Skipping due to state=$(_jq ${pr} '.state') and draft=$(_jq ${pr} '.draft')."
+    continue
+  fi
+
+  ref=$(_jq ${pr} '.base.ref')
+  sha=$(_jq ${pr} '.base.sha')
+
+  shaToCompare=${listSha[${ref}]}
+
+  # Keep the list of branch head sha
+  if [[ $shaToCompare == "" ]]
+  then
+    refCommit=$(curl -s \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: token $oauth" \
+              https://api.github.com/repos/imodeljs/imodeljs/commits/${ref})
+    refHeadCommit=$(echo ${refCommit} | jq -r '. | @base64')
+    refHeadSha=$(_jq  ${refHeadCommit} '.sha')
+    listSha[${ref}]=${refHeadSha}
+    shaToCompare=$refHeadSha
+  fi
+
+  # Compare base sha with the target branch head sha
+  log "  Head sha at ${ref} branch is ${shaToCompare} and base sha is ${sha}"
```

Comment:
Instead of `head sha` can you make this `Target branch, ${ref}, commit sha is ${shaToCompare} and PR is on sha ${sha}`?  Or something along those lines just to make it more clear that it is the target branch that is further ahead.